lab_3_generate_by_ngrams/main.py

```diff
@@ -270,6 +403,27 @@ def run(self, seq_len: int, prompt: str) -> Optional[str]:
         In case of corrupt input arguments or methods used return None,
         None is returned
         """
+        if not isinstance(seq_len, int) or not isinstance(prompt, str) or len(prompt) == 0:
+            return None
+
+        n_gram_size = self._model.get_n_gram_size()
+        encoded = self._text_processor.encode(prompt)
+        if not encoded or not n_gram_size:
+            return None
+
+        while seq_len > 0:
+            candidates = self._model.generate_next_token(encoded[-n_gram_size + 1:])
+            if not candidates:
+                break
+            best_candidate = [letter for letter, freq in candidates.items() if freq == max(candidates.values())]
```

Comment:
filter and get `max(candidates.values())` only once

Reply:
It didn't work out